src/hotspot/share/logging/logFileStreamOutput.cpp

```diff
@@ -146,6 +164,28 @@ int LogFileStreamOutput::write_internal(const LogDecorations& decorations, const
   return written;
 }
 
+int LogFileStreamOutput::write_internal(const LogDecorations& decorations, const char* msg) {
+  int msg_len = (int)strlen(msg);
```

Comment:
Not OK with casting this. Add some sort of overflow check instead and only then `static_cast`.